src/geometry/broad_phase_multi_sap.rs

```diff
@@ -640,11 +651,18 @@ impl BroadPhase {
         }
     }
 
+    fn update_regions(&mut self) {
```

Comment:
Just a thought about that: it seems that currently as soon as `deleted` > 0, you go through all regions to check which needs update and which are to be discarded. It seems a good start, but isn't there a scalability issue for a large world in which many regions are not having any change, and a small set are changed a lot? An alternative could be to collect regions that need update or deletion, and touch only these. As I do not have the full design in mind, maybe there is a trivial reason why that would not work, but when I saw this code, this question came to my mind, so I wrote it down here, in case it can be useful.

Reply:
Unless I am misunderstanding you, I don't believe this change adds a major scalability issue, at least from an algorithmic viewpoint. The previous version and the version with my changes both iterate over all the regions, which is of course `O(#of regions)`. Perhaps you mean that we could avoid making multiple passes? This is probably correct, but it would require a mutate-while-iterating pattern which is usually tricky to do in Rust.

Reply:
Indeed your change does not change complexity, I put a note here because while reading this code I realised this potential scalability problem, that was present already. Maybe I should have commented the original code instead.

What came to my mind for this note was that, assuming we only process objects that are awake, if a large world has a lot of objects and only a subset is awake, we still process all regions. This is in contrast to a tree structure that keeps note whether there are any awake object in its sub-tree, that could have some better scalability properties. But it seems that it is an optimisation for later probably...

Reply:
Regarding the double loop, we could avoid it by storing the points (keys) of regions with 0 proxies. I am not sure that would be more performant, I guess it depends on the ratio of regions with 0 proxies compared to the total number of regions.

Reply:
Ah yes! Sorry I misunderstood. That's definitely a good thought in my opinion.